packages/ui/src/containers/AppContent/index.tsx

```diff
@@ -15,6 +23,17 @@ export type Props = {
 const AppContent = ({ children }: Props) => {
   const theme = useTheme();
   const { toast, platform, setToast, experienceSettings } = useContext(PageContext);
+  const [topSpace, setTopSpace] = useState(0);
+  const topPlaceHolderRef = useRef<HTMLDivElement>(null);
+
+  useLayoutEffect(() => {
+    setTimeout(() => {
+      if (!topPlaceHolderRef.current) {
+        return;
+      }
+      setTopSpace(topPlaceHolderRef.current.offsetHeight);
+    });
+  }, [topPlaceHolderRef]);
```

Comment:
this looks very tricky. @wangsijie @demonzoo any ideas?

Reply:
The requirement itself is tricky, IMO.

Reply:
let's schedule a quick meeting on Mon